### PR TITLE
Document WithQuery extension method

### PR DESCRIPTION
**Pull request by Simply007**
### Motivation

Internal link: https://github.com/Kentico/statiq-kontent-collaboration/issues/16

WithQuery was hard to find (I had to dive into the implementation to find it).

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,26 @@ You should now see that for every Article in the Kontent site there's an html fi
...
+
+If you need to filter out the the input document s from Kontent, it is possible to specify the query of your request using `WithQuery(IQueryParameter[])` method.
+
+Let's say, you just want first three latest articles that have the Title element set and you want to load only a subset of.
+
+```csharp
+// ...
+new Kontent<Article>(client)
+    .WithQuery(
+        new NotEmptyFilter("elements.title"),
```
Comment (alanta) [inline README.md]: The model generator provides all the strings used in this query. As a coding best practice I'd use those strings as much as possible

Comment (Simply007) [reply README.md]: Do you mean something like that? Or is there any other way to omit the "elements" part?

```suggestion
        new NotEmptyFilter("$"elements.{Article.TitleCodename}""),
```

Comment (alanta) [reply README.md]: Yes, that's what I meant. @petrsvihlik is there a more elegant way to handle the "elements" part?

Comment (petrsvihlik) [reply README.md]: This is currently the best way to do it...I tried to outline a solution here: https://github.com/Kentico/kontent-delivery-sdk-net/issues/241#issuecomment-724751872
Please comment.

**Pull request merged**